Allegro 4.4.3: Malformed UTF-8 character (fatal)

Building `allegro` with `nix-shell -p allegro` (on nixpkgs unstable) leads to

```sh
utf8 "\xB0" does not map to Unicode at /nix/store/jjnl8b9gd6ri1bwfrkcbsz169qhj7asb-texinfo-6.7/share/texinfo/Texinfo/ParserNonXS.pm line 1796, <FH> line 18520.
Malformed UTF-8 character: \xb0 (unexpected continuation byte 0xb0, with no preceding start byte) in pattern match (m//) at /nix/store/jjnl8b9gd6ri1bwfrkcbsz169qhj7asb-texinfo-6.7/share/texinfo/Texinfo/ParserNonXS.pm line 3364.
Malformed UTF-8 character (fatal) at /nix/store/jjnl8b9gd6ri1bwfrkcbsz169qhj7asb-texinfo-6.7/share/texinfo/Texinfo/ParserNonXS.pm line 3364.
make[2]: *** [docs/CMakeFiles/docs.dir/build.make:435: docs/info/allegro.info] Error 25
make[1]: *** [CMakeFiles/Makefile2:647: docs/CMakeFiles/docs.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
I found [this](https://bugzilla.redhat.com/show_bug.cgi?id=1767827) identical issue from 2019 in RedHat.

They claim this patch
```patch
diff -up allegro-4.4.3/docs/src/allegro._tx.orig allegro-4.4.3/docs/src/allegro._tx
--- allegro-4.4.3/docs/src/allegro._tx.orig     2019-02-02 20:28:46.000000000 +0100
+++ allegro-4.4.3/docs/src/allegro._tx  2019-11-04 11:12:39.352699777 +0100
@@ -23,6 +23,7 @@
 @man_shortdesc_force1=allegro
 @man_shortdesc_force2=Allegro game programming library.
 @$\input texinfo
+@$@documentencoding ISO-8859-1
 @$@setfilename allegro.inf
 @$@settitle Allegro Manual
 @$@setchapternewpage odd
```
fixes the issue, however I haven't tested this. (Mainly because I don't know how)

I'd be happy to try to fix this, looks like just adding that patch will do the trick, but I'm new to nixpkgs

**Notify maintainers**
@chkno @7c6f434c

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.54, NixOS, 20.09pre236721.840c782d507 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(jakob): `""`
 - channels(root): `"home-manager, nixos-20.09pre242769.61525137fd1, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/allegro/default.nix
+++ b/pkgs/development/libraries/allegro/default.nix
@@ -14,6 +14,7 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./nix-unstable-sandbox-fix.patch
+    ./encoding.patch
   ];
 
   buildInputs = [

--- a/pkgs/development/libraries/allegro/encoding.patch
+++ b/pkgs/development/libraries/allegro/encoding.patch
@@ -1,0 +1,10 @@
+--- allegro-4.4.3/docs/src/allegro._tx.orig     2019-02-02 20:28:46.000000000 +0100
++++ allegro-4.4.3/docs/src/allegro._tx  2019-11-04 11:12:39.352699777 +0100
+@@ -23,6 +23,7 @@
+ @man_shortdesc_force1=allegro
+ @man_shortdesc_force2=Allegro game programming library.
+ @$\input texinfo
++@$@documentencoding ISO-8859-1
+ @$@setfilename allegro.inf
+ @$@settitle Allegro Manual
+ @$@setchapternewpage odd

